git_big/main.py

```diff
@@ -362,8 +372,8 @@ def get(self, entry):
         cache_dir = os.path.dirname(entry.cache_path)
         if not os.path.exists(cache_dir):
             os.makedirs(cache_dir)
-        os.rename(pending_path, entry.cache_path)
```

Comment:
Could you explain the reason for changing the order here? There is some refcounting that I thought I was exploiting here (on POSIX).